File: OOP/Class.js
class Animal {
  constructor(name, sound) {
    this.name = name;
    this.sound = sound;
  }

  speak() {
    return `${this.name} says ${this.sound}`;
  }

  toString() {
    return `Animal(${this.name})`;
  }
}

const cat = new Animal("Cat", "Meow");
console.log(cat.speak());    // "Cat says Meow"
console.log(`${cat}`);       // "Animal(Cat)" — uses toString()

File: OOP/Class.py
class Animal:
    def __init__(self, name, sound):
        self.name = name
        self.sound = sound

    def speak(self):
        return f"{self.name} says {self.sound}"

    def __str__(self):
        return f"Animal({self.name})"


cat = Animal("Cat", "Meow")
print(cat.speak())
print(f"{cat}")
